guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,927 @@
+so Diesel offers custom derives.
+
+In this guide,
+we will cover each trait in detail in terms of its use cases 
+and usage considerations.
+To make use of these derives,
+you will only need `#[macro_use] extern crate diesel;` at the root of your project.
+
+Throughout this guide,
+we will be looking at examples for each trait and how they interact together.
+Some of the example code will be implementing basic [CRUD] database operations.
+We'll be covering creating, reading, and updating data.
+The details of those operations will be not be covered beyond their relevance to the demonstrated trait.
+
+In general, it may be more helpful to think of Diesel as a SQL query builder.
+While Diesel does offer some standard ORM (Object Relation Mapper) features,
+Diesel's code generation derives are for safely building SQL queries.
+
+[CRUD]: https://en.wikipedia.org/wiki/Create,_read,_update_and_delete
+
+- [Queryable](#queryable)
+- [QueryableByName](#queryablebyname)
+- [Insertable](#insertable)
+- [Identifiable](#identifiable)
+- [AsChangeset](#aschangeset)
+- [Associations](#associations)
+
+## Queryable
+A `Queryable` struct is one that represents the 
+data structure returned from a database query.
+In many cases this may map exactly to the column structure of a single table,
+however there may be cases where you need to make a query that spans several tables and/or only uses
+a subset of columns. 
+For this reason, it may be helpful to view `Queryable` structs as the *result* of your query.
+It is acceptable and often desirable to have multiple `Queryable` structs for the same database table.
+
+Annotating your struct with [`#[derive(Queryable)]`][queryable_doc] enables you to use Diesel's
+`RunQueryDsl` to assist in retrieving data.
+A few of the methods you may use are `load()`, `get_result()`, `get_results()`, and `first()`.
+Should you make a query that doesn't return the same columns and values (in the order specified) on your `Queryable` struct,
+you will get a compile-time error.
+The only thing `Queryable` cares about is that the data returned from the query
+maps exactly to your data structure.
+
+[queryable_doc]: https://docs.diesel.rs/diesel/query_source/trait.Queryable.html
+
+The following example shows making two different queries into the `users` table.
+We get back a [`QueryResult`],
+which is basically a wrapper around the rust `Result` type.
+That means we'll be able to use `expect()` to handle our error conditions.
+
+[`QueryResult`]: https://docs.diesel.rs/diesel/result/type.QueryResult.html
+
+### Example
+
+```rust
+// File: src/models.rs
+
+use schema::users; // Brings the users table into scope
+
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+#[derive(Queryable)]
+pub struct EmailUser {
+    pub id: i32,
+    pub email: String,
+}
+```
+
+```rust
+// File: src/main.rs
+
+use diesel::prelude::*;
```

Comment:
You told people to add `#[macro_use] extern crate diesel;`, so maybe add it here?